keps/sig-auth/267-kubelet-server-certificate-bootstrap-rotation/README.md

```diff
@@ -0,0 +1,438 @@
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This is retrospective KEP description for the feature `RotateKubeletServerCertificate`.
+
+The feature gate `RotateKubeletServerCertificate` allows to configure kubelet to
+get a server certificate for the kubelet from the Certificate Signing Request
+API instead of generating one self signed and auto rotates the certificate as
+expiration approaches.
+
+## Motivation
+
+The feature is heavily used and is a part of [CIS Kubernetes benchmark](https://www.cisecurity.org/benchmark/kubernetes):
+
+> 1.3.6 Ensure that the `RotateKubeletServerCertificate` argument is set to true
+
+This indicates feature maturity and high production readiness.
+
+The outstanding work of generalizing the approach certificate requests got
+approved will require a new KEP.
+
+
+### Goals
+
+- ability to configure kubelet to get a server certificate for the kubelet from
+  the Certificate Signing Request API instead of generating one self signed
+- auto-rotate the certificate nearing expiration using Certificate Signing
+  Request API
+
+### Non-Goals
+
+- built-in support for approving CSRs nodes request (not currently done because
+  there is not a way to determine if a node controls a given IP or DNS name)
+- ability to limit the addresses/hostnames a node requests in its cert
+  (currently, nodes request all addresses reported by the cloud provider, and
+  the selection of which address to use is done on the API server side)
+
+## Proposal
+
+Documentation for the feature can be found here:
+
+https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#certificate-rotation
+
+### Notes/Constraints/Caveats (Optional)
+
+The note about default implementation that will not approve certificate requests
+is already a part of documentation. See the Note on this page: https://kubernetes.io/docs/reference/access-authn-authz/kubelet-tls-bootstrapping/#certificate-rotation
+
+### Risks and Mitigations
+
+The feature was in heavy production use for a while. Likely no risks will emerge
+with locking the feature gate.
+
+## Design Details
+
+Here are limited design definition for the retrospective KEP.
+
+Implementation PRs:
+- Feature gate introduced: [#45059](https://github.com/kubernetes/kubernetes/pull/45059)
+- Feature promoted to GA: [#51045](https://github.com/kubernetes/kubernetes/pull/51045)
```

Comment:
I recommend adding a bit more detail. These just link to code. The full details might not be needed but we should write up more. That will also help people who are reviewing the docs for the GA graduation.